doc/api/buffer.markdown

```diff
@@ -541,6 +541,42 @@ console.log(bufA.length);
 // 42
 ```
 
+### Class Method: Buffer.encode(str[, str_encoding][, target_encoding])
```

Comment:
this is mind bending. collapsing optional arguments in JS is almost always based on type. since both these arguments are strings it depends on the number of arguments passed. not sure I agree with this.

the way the arguments are stated, `str_encoding` and `target_encoding` would need to be different types for us to know how they collapse. @addaleax is correct that the brackets need to be changed to properly signify which one is processed first.